### ISSUE TITLE
Issues with sc.pp.highly_variable_genes following tutorial

### ISSUE THREAD

**Issue by oligomyeggo** [theislab/single-cell-tutorial#32]
This is related to closed issue #8, so I apologize for the repetition. Since the issue was closed I wasn't sure if I could comment or not.

Essentially, I am following your excellent and thorough Case-study_Mouse-intestinal-epithelium_1906.ipynb notebook, but using my own data. After the Normalization steps (and skipping the Batch Correction section, as I only have one sample), I am unable to run the `sc.pp.highly_variable_genes` command in In[26]. I receive the following error: 
`LinAlgError: Last 2 dimensions of the array must be square`. However, if I omit the scran normalization process (In[18]-In[21]) and replace In[23] with: 

```
sc.pp.normalize_total(adata, target_sum=1e4)
sc.pp.log1p(adata)
```

I can execute the rest of the code. Is this an OK practice to do, or is scran normalization much better? And is scran normalization is better, how can I fix my error?

Also, perhaps related (?), I am trying to run the R tradeSeq package (https://github.com/statOmics/tradeSeqPaper) downstream of the slingshot object generated following the steps in your tutorial. However, I keep erroring out on their `fitGAM` function, receiving the following error: 
`could not broadcast input array from shape (2) into shape (1136)`. This makes me feel like maybe there is something wrong with my adata object somewhere along the way, although I was able to run other R packages like Slingshot with no issue. Any insight/idea how to run tradeSeq downstream of Slingshot following data processing as outlined in your tutorial?

Thank you!

**Comment by LuckyMD** [comment]
Hi @oligomyeggo, 

So counts per million normalization has a large assumption that doesn't hold true for most single-cell datasets. It assumes that all cells have equal mRNA molecules and such you can use relative proportions to accurately compare cells. The [scran pooling paper](http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7) does quite a good job at explaining this.

The issue is probably that using scran pooling gives you a dense matrix. This can be turned back to sparse by running `adata.X = scipy.sparse.csr_matrix(adata.X)` as suggested in #8. Could you check the output of `type(adata.X)` at various points of your script? The output should never be `np.matrix`.

As for running the R tradeSeq package, I'm afraid I can't really help you there as I'm unfamiliar with the code. I would report the issue on their github and see what they say. Just so you're aware, the object you have when you are running R code isn't actually an `AnnData` object anymore, but a `SingleCellExperiment` R object. With `anndata2ri` two objects can have the same name, but one is in python as an `AnnData` object and the other is in R as a `SingleCellExperiment` object. If you use it downstream of `Slingshot` you might even have a `SlingshotObject`. Note that you cannot move a `SlingshotObject` back into python without loss of information at the moment.

Hope this helps!

**Comment by oligomyeggo** [comment]
Yes, that helped. Thank you! I apologize. I had misread issue #8 and thought that I should not be running `adata.X = scipy.sparse.csr_matrix(adata.X)`. I ran that line after the `# Normalize adata` chunk as suggested in #8. Right after the `# Normalize adata` `type(adata.X) = numpy.matrix`. After converting the matrix, `type(adata.X) = scipy.sparse.csr.csr_matrix`, and I have been able to continue through the workflow without error.

As for `tradeSeq` I figured that issue out as well (sort of, after getting help on their github). Even though I have had no issues going back and forth between python `AnnData` objects and R `SingleCellExperiment` objects using packages outlined in your tutorial, for some reason `tradeSeq` was not happy with my objects generated from `Slingshot` in jupyter notebook. However, if I saved the objects needed for `tradeSeq` and opened them in RStudio I was able to continue with the analysis. I'm sure there is a way to get `tradeSeq` working in jupyter notebook, and maybe I just configured by R chunks wrong by mistake. At any rate, I've had no issue moving forward with that part of my analysis in RStudio. Just thought I would share in case anyone else is trying to do what I'm doing and runs into the same issue.

Thank you for your help!
